cluster/charts/crossplane-types/crds/core.oam.dev_containerizedworkloads.yaml

```diff
@@ -68,6 +68,25 @@ spec:
                       description: A ContainerConfigFile specifies a configuration
                         file that should be written within a container.
                       properties:
+                        fromSecret:
+                          description: FromSecret is a secret key reference which
+                            can be used to assign a value to be written to the configuration
+                            file at the given path in the container.
+                          properties:
+                            key:
+                              description: The key to select.
+                              type: string
+                            name:
+                              description: Name of the secret.
+                              type: string
+                            namespace:
+                              description: Namespace of the secret.
+                              type: string
+                          required:
+                          - key
+                          - name
+                          - namespace
```

Comment:
Our `SecretKeySelector` implementation requires the namespace to be specified. In OAM we presumably want to load the secret from the workload's namespace. We could use `corev1.SecretKeySelector`, but that embeds `corev1.LocalObjectReference`, which has a (strangely) optional `Name` key. I'd suggest we add a namespaceless `SecretKeySelector` implementation to the OAM type definitions.

Reply:
@negz good point. I am now wondering how the tests I ran were successful 🤔

Reply:
For anyone looking back at this, the issue I encountered was that I had provided a `secretFrom` field rather than a `fromSecret` field. This was not pruned because of https://github.com/kubernetes-sigs/controller-tools/pull/362